tests/unit/multiplexer_test.py

```diff
@@ -43,3 +44,16 @@ def problematic_iterator():
 
         with self.assertRaises(Problem):
             list(mux.loop())
+
+    def test_cascade_stop(self):
+        mux = Multiplexer([
+            ((lambda x: sleep(0.01) or x)(x) for x in ['after 0.01 sec T1', 'after 0.02 sec T1', 'after 0.03 sec T1']),
+            ((lambda x: sleep(0.02) or x)(x) for x in ['after 0.02 sec T2', 'after 0.04 sec T2', 'after 0.06 sec T2']),
+        ], cascade_stop=True)
```

Comment:
It would be great to test this without sleeping, but then I guess a failure would freeze the test suite.

Could we have one stream without any sleeps, and another that emits one event then sleeps for 0.01?  That way we should be sleeping at most just the 0.1s, and the one that doesn't sleep should always finish before that.